### DISCUSSION TITLE
Support `HTTPMethod` for `@api_view`

### DISCUSSION THREAD

**Discussion by johnthagen** [encode/django-rest-framework#9397] [Ideas & Suggestions]
The `@action` [docs support using Python 3.11 `HTTPMethod`](https://www.django-rest-framework.org/api-guide/viewsets/#marking-extra-actions-for-routing) rather than bare strings.

Could this be extended to `@api_view`, where I think it would have similar use

- https://www.django-rest-framework.org/api-guide/views/#api_view

```py
@api_view([HTTPMethod.GET, HTTPMethod.POST])
def hello_world(request):
    if request.method == HTTPMethod.POST:
        ...
```

Related to 

- #8995

**Comment by lovelydinosaur** [comment]
I think it should either be documented / supported in both places or neither.

Personally I'd probably suggest the second, and *dropping* it from the API docs. No problem with folks using the `HTTPMethod.GET` style if they want to use that enum, but there's probably a whole lot of places we could be switching documentation around if we start taking that approach.

(That'd be my first take anyways... ?)

**Reply by johnthagen** [reply]
I personally prefer `Enum`s whenever possible because editors provide auto complete and can catch typos quickly in real time. Bare strings are easier to miss typos on, and if you aren't using `djangorestframework-stubs`, there's not much of a way to catch these statically.

**Reply by lovelydinosaur** [reply]
Okay, well let's start with this...

> Could this be extended to @api_view, where I think it would have similar use

Does `HTTPMethod.GET` currently work for `api_view`?

**Reply by johnthagen** [reply]
I haven't tried since I don't know if that is a supported/intended use case. In the case of #8995, tests were required to ensure that it is in fact supported and does not regress 

- https://github.com/encode/django-rest-framework/pull/9067/files

**Reply by auvipy** [reply]
It would be great if you come with some test case atleast....

**Reply by johnthagen** [reply]
@tomchristie 

> Does `HTTPMethod.GET` currently work for `api_view`?

I replaced bare `str` literals (included examples with `"GET"`, `"PATCH"`, and `"POST"`) with `HTTPMethod` across 6 `api_view`s in my DRF application. My entire unit test suite passed and the OpenAPI schema generated by `drf-spectacular` remained unchanged.

So it appears that this works currently.

> It would be great if you come with some test case atleast....

Sorry, based on the discussion I wasn't sure if this was even something desired by the DRF maintainers.

**Reply by lovelydinosaur** [reply]
Thanks @johnthagen. It's a perfectly reasonable question.
I'm not that keen on it myself.

Probably a more comprehensive question here is where are *all* the cases of using HTTP methods as string literals (throughout) and are all the cases where we use them as string literals case insensitive or not?
